include/head/core/presence_sensing_driver.hpp

```diff
@@ -26,8 +96,42 @@ class PresenceSensingDriver {
         return voltage_read;
     }
 
+    auto static get_tool(adc::MillivoltsReadings reading) -> AttachedTool {
+        auto at = AttachedTool{.z_motor = can_ids::ToolType::UNDEFINED_TOOL,
+                               .a_motor = can_ids::ToolType::UNDEFINED_TOOL,
+                               .gripper = can_ids::ToolType::UNDEFINED_TOOL};
+        for (auto& element : OT3ToolList) {
+            if ((reading.z_motor < element.detection_upper_bound) &&
+                (reading.z_motor >= element.detection_lower_bound) &&
+                (element.tool_carrier == Z_CARRIER)) {
+                at.z_motor = element.tool_type;
+            }
+            if ((reading.a_motor < element.detection_upper_bound) &&
+                (reading.a_motor >= element.detection_lower_bound) &&
+                (element.tool_carrier == A_CARRIER)) {
+                at.a_motor = element.tool_type;
+            }
+            if ((reading.gripper < element.detection_upper_bound) &&
+                (reading.gripper >= element.detection_lower_bound) &&
+                (element.tool_carrier == UNDEFINED_CARRIER)) {
+                at.gripper = element.tool_type;
+            }
+        }
+
+        return at;
+    }
+
+    auto getCurrentTools() -> AttachedTool { return this->current_tools; }
```

Comment:
these are `camelCase` but `get_tool` and `get_readings` are `snake_case`. let's just do `snake_case` since that's what the rest of hte project uses.